src/Files.App/Actions/FileSystem/EmptyRecycleBinAction.cs

```diff
@@ -1,19 +1,52 @@
-﻿using Files.App.Commands;
+﻿using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.DependencyInjection;
+using Files.App.Commands;
+using Files.App.Contexts;
 using Files.App.Extensions;
 using Files.App.Helpers;
+using System.ComponentModel;
 using System.Threading.Tasks;
 
 namespace Files.App.Actions
 {
-	internal class EmptyRecycleBinAction : IAction
+	internal class EmptyRecycleBinAction : ObservableObject, IAction
 	{
+		private readonly IContentPageContext context = Ioc.Default.GetRequiredService<IContentPageContext>();
+
 		public string Label { get; } = "EmptyRecycleBin".GetLocalizedResource();
 
 		public RichGlyph Glyph { get; } = new RichGlyph("\uEF88", fontFamily: "RecycleBinIcons");
 
+		public bool IsExecutable
+		{
+			get
+			{
+				if (context.PageType is ContentPageTypes.RecycleBin)
+					return context.HasItem;
+				return RecycleBinHelpers.RecycleBinHasItems();
```

Comment:
Can we just use a simple `RecycleBinHelpers.RecycleBinHasItems()` here? If you did it, I suppose it's because the `context.hasItem` is lighter to execute?

Reply:
Yes, context.HasItem is 5-10 times faster. But this is a trick that can only be used if the RecycleBin is displayed in the current page.